db/compaction/compaction_job.cc

```diff
@@ -914,6 +916,11 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
   }
   const auto& c_iter_stats = c_iter->iter_stats();
 
+  std::unique_ptr<SstPartitioner> partitioner =
+      sub_compact->compaction->CreateSstPartitioner();
+  SstPartitioner* partitioner_ptr = partitioner.get();
```

Comment:
why adding a new variable instead of calling `partitioner->XXX()` directly?

Reply:
Right, fixed.